File: test/tests/chained.js
const { assert } = require('chai');
const { mocha_object_store } = require('../opts');

it('nominal', () => {
  const alice = new dll.Chained;
  assert.instanceOf(alice, dll.Chained);
  assert.strictEqual(alice.inc1().inc10().inc100().get(), 111);
  assert.strictEqual(alice.get(), 111);

  // Object store
  if (!mocha_object_store())
    return;
  assert.strictEqual(alice.inc1().inc10().inc100(), alice);
});

File: test/tests/chained.cc
#include <fixtures/chaining.h>

#include <nobind.h>

NOBIND_MODULE(chained, m) {
  m.def<Chained>("Chained")
    .cons<>()
    .def<&Chained::Inc1, Nobind::ReturnShared>("inc1")
    .def<&Chained::Inc10, Nobind::ReturnShared>("inc10")
    .def<&Chained::Inc100, Nobind::ReturnShared>("inc100")
    .def<&Chained::Get>("get");
}
